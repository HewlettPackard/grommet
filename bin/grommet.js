#!/usr/bin/env node

var gulp = require('gulp');
var template = require('gulp-template');
var install = require('gulp-install');
var path = require('path');
var mkdirp = require('mkdirp');
var shelljs = require('shelljs');

String.prototype.capitalize = function() {
  var words = this.split(' ');

  words = words.map(function(word) {
    return word.charAt(0).toUpperCase() + word.slice(1).toLowerCase();
  });

  return words.join(' ');
};

var task = process.argv[2] || 'init';
var app = process.argv[3] || 'app-name';
var title = app.replace(/-|_/g, ' ').capitalize();
var grommetPath = path.join(__dirname, '..');

function nodeVersionSupported() {
  return Number(process.version.match(/^v(\d+\.\d+)/)[1]) >= 0.10;
}

var npmVersion;
function npmVersionSupported() {
  npmVersion = Number(shelljs.exec('npm --version').output.toString().match(/^(\d+\.\d+)/)[1]);
  return npmVersion  >= 1.4;
}

gulp.task('init', function(done) {

  if (!nodeVersionSupported() || !npmVersionSupported()) {
    console.error('[grommet] Grommet requires Node v0.10+ and NPM 1.4.x+.');
    console.error('[grommet] Currently you have Node ' + process.version + ' and NPM ' + npmVersion);
    process.exit(1);
  }

  mkdirp('./' + app, function(err) {
    if (err) {
      console.log('Error trying to create project: ' + err);
    } else {
      process.chdir('./' + app);

      var templateFolder = path.join(grommetPath, 'templates/init/**');
      var mobileIcon = path.join(grommetPath, 'mobile-app-icon.png');
      var shortcutIcon = path.join(grommetPath, 'shortcut-icon.png');

      var packageJSON = require(path.join(grommetPath, 'package.json'));

      gulp.src(mobileIcon).pipe(gulp.dest('./src/img'));
      gulp.src(shortcutIcon).pipe(gulp.dest('./src/img'));
      gulp.src(templateFolder, { dot: true })
      .pipe(template({
        appName: app,
        appTitle: title,
        grommetVersion: packageJSON.version
      }))
      .pipe(gulp.dest('./'))
      .pipe(install()).on('finish', function() {
        done();
      });
    }
  });

});

gulp.task('export', function() {
  console.log('[REMOVED] All the example apps now live outside Grommet. Checkout Grommet organization in Github to learn more (https://github.com/grommet).');
});

var argv = require('yargs').argv;

if (argv.version) {
  console.log(require(path.join(grommetPath, 'package.json')).version);
  process.exit(0);
} else {
  gulp.start(task);
}
