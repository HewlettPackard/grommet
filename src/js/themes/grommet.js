import { css } from 'styled-components';

import { deepFreeze } from '../utils';

const workSansPath = 'https://fonts.gstatic.com/s/worksans/v2';
const zillaSlabPath = 'https://fonts.gstatic.com/s/zillaslab/v3';

export const grommet = deepFreeze({
  global: {
    font: {
      family: "'Work Sans', Arial, sans-serif",
      face: `
        @font-face {
          font-family: 'Work Sans';
          font-style: normal;
          font-weight: 400;
          src:
            local('Work Sans'),
            local('WorkSans-Regular'),
            url("${workSansPath}/ElUAY9q6T0Ayx4zWzW63VJBw1xU1rKptJj_0jans920.woff2") format('woff2');
        }

        @font-face {
          font-family: 'Work Sans';
          font-style: normal;
          font-weight: 600;
          src:
            local('Work Sans SemiBold'),
            local('WorkSans-SemiBold'),
            url("${workSansPath}/z9rX03Xuz9ZNHTMg1_ghGRampu5_7CjHW5spxoeN3Vs.woff2") format('woff2');
        }

        @font-face {
          font-family: 'Zilla Slab';
          font-style: normal;
          font-weight: 400;
          src:
            local('Zilla Slab Regular'),
            local('ZillaSlab-Regular'),
            url("${zillaSlabPath}/dFa6ZfeM_74wlPZtksIFajo6_V6LVlA.woff2") format('woff2');
        }

        @font-face {
          font-family: 'Zilla Slab';
          font-style: normal;
          font-weight: 600;
          src:
            local('Zilla Slab SemiBold'),
            local('ZillaSlab-SemiBold'),
            url("${zillaSlabPath}/dFa5ZfeM_74wlPZtksIFYuUe6HOpW3pwfa0.woff2") format('woff2');
        }
      `,
    },
  },
  anchor: {
    textDecoration: 'underline',
    color: {
      dark: '#9060EB',
      light: '#9060EB',
    },
  },
  button: {
    extend: css`
      ${props => !props.plain && 'font-weight: bold;'}
    `,
  },
  checkBox: {
    icon: {
      extend: css`
        box-sizing: border-box;
        position: absolute;
        top: 0px;
        left: 0px;
        width: ${props => props.theme.checkBox.size};
        height: ${props => props.theme.checkBox.size};
      `,
    },
  },
  heading: {
    font: {
      family: "'Zilla Slab', 'Work Sans', Arial, sans-serif",
    },
  },
});
