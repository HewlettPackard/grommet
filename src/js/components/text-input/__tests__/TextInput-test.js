import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { Grommet } from '../../grommet';
import { TextInput } from '../';

test('TextInput renders', () => {
  const component = renderer.create(
    <Grommet>
      <TextInput id='item' name='item' />
    </Grommet>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
