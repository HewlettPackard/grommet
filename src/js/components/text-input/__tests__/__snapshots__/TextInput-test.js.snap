// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders 1`] = `
.c0 {
  font-family: 'Work Sans',Arial,sans-serif;
  font-size: 1em;
  line-height: 1.5;
  color: #333333;
  background-color: #FFFFFF;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c0 * {
  box-sizing: inherit;
}

.c1 {
  padding: 11px;
  border: 1px solid rgba(0,0,0,0.15);
  border-radius: 4px;
  outline: none;
  background-color: transparent;
  color: inherit;
  font: inherit;
  margin: 0;
}

.c1::-webkit-input-placeholder {
  color: #AAAAAA;
}

.c1::-moz-placeholder {
  color: #AAAAAA;
}

.c1:-ms-input-placeholder {
  color: #AAAAAA;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

.c1:focus {
  border-color: #00CCEB;
  box-shadow: 0 0 1px 1px #00CCEB;
}

@media only screen and (min-width:481px) {
  .c0 {
    top: 0px;
    bottom: 0px;
    left: 0px;
    right: 0px;
    height: 100%;
    width: 100%;
    overflow: visible;
  }
}

<div
  className="c0"
>
  <input
    autoComplete="off"
    className="c1"
    defaultValue={undefined}
    id="item"
    name="item"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    value={undefined}
  />
</div>
`;
